resteasy-jaxrs/src/main/java/org/jboss/resteasy/core/registry/SegmentNode.java

```diff
@@ -101,7 +101,23 @@ public ResourceInvoker match(HttpRequest request, int start)
                }
                else
                {
-                  String substring = path.substring(0, length);
+                  // must find the end of the matched pattern
+                  // and get the substring from 1st char thru end
+                  // of matched chars
+                  Pattern p = expression.getPattern();
+                  Matcher m = p.matcher(path);
+                  m.region(start, path.length());
+                  String substring = path;
+                  while(m.find()) {

```

Comment:
Should this be a while? Wouldn't the final (/.+)? gobble up the rest of the path, so that the entire path gets matched and there can never be a second successful m.find()?  Well, maybe there's some strange regex such that that is not the case, but, even so, wouldn't you want to take the first such match?